File: project_euler_problems/solved_problems/c_solutions/problem_3/3.c
#include <stdio.h>

/* Project Euler #3:
 * The prime factors of 13195 are 5, 7, 13 and 29.
 *
 * What is the largest prime factor of the number 600851475143?
 */

int main()
{
    long number = 600851475143;
    long largest_prime_factor = 1;
    long divisor = 2;
    while (number > 1) {
        if (number % divisor == 0) {
            largest_prime_factor = divisor;
            number = number / divisor;
            while (number % divisor == 0) {
                number = number / divisor;
            }
        }
        divisor++;
    }
    // Answer: 6857
    printf("The largest prime factor: %ld\n", largest_prime_factor);
    return 0;
}

File: project_euler_problems/solved_problems/python_solutions/3.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

"""
Project Euler Problem #3:

The prime factors of 13195 are 5, 7, 13 and 29.
What is the largest prime factor of the number 600851475143 ?
"""

number = 600851475143
x = 2
ans = 1
while number > 1:
	if number % x == 0:
		ans = x
		number = number/x
		while number % x == 0:
			number = number/x
	x += 1

print "Maximum Prime Factor: %d" % ans #Answer: 6857
